### PR TITLE
Update wpilib.drive and fix docs

### PR DESCRIPTION
**Pull request by auscompgeek**
* Fix the broken drive base diagrams.
* Put the RobotDrive porting guides in note directives for clarity.
* Add the missing code snippets in DifferentialDrive.
* [upstream change] Invert the right motor on LiveWindow in DifferentialDrive.

### CODE REVIEW

File: wpilib/wpilib/drive/differentialdrive.py [hunk 1]
```diff
@@ -70,13 +101,13 @@ class DifferentialDrive(RobotDriveBase):
     def __init__(self, leftMotor, rightMotor):
         """Constructor for DifferentialDrive.
 
-        .. note:: This class takes in a SpeedController per side. For two and four motor drivetrains,
-                  construct and pass in  :class:`.SpeedControllerGroup` instances as follows.
+        .. note:: To pass multiple motors per side, use a :class:`.SpeedControllerGroup`.
```
Comment (twinters007) [inline wpilib/wpilib/drive/differentialdrive.py]: While your docs are better than upstreams, I think we should keep upstreams

File: wpilib/wpilib/drive/differentialdrive.py [hunk 2]
```diff
@@ -70,13 +101,13 @@ class DifferentialDrive(RobotDriveBase):
...
+                  If a motor needs to be inverted, do so before passing it in.
 
-        :param leftMotor: Left motor SpeedController
-        :param rightMotor: Right motor SpeedController
-        :type leftMotor: :class:`.SpeedController` or :class:`.SpeedControllerGroup`
-        :type rightMotor: :class:`.SpeedController` or :class:`.SpeedControllerGroup`
+        :param leftMotor: Left motor(s)
+        :param rightMotor: Right motor(s)
+        :type leftMotor: :class:`.SpeedController`
```
Comment (twinters007) [inline wpilib/wpilib/drive/differentialdrive.py]: Why get rid of the or? I guess a SpeedControllerGroup is a speed controller...But it makes it clearer in the docs imo

File: wpilib/wpilib/drive/differentialdrive.py [hunk 3]
```diff
@@ -95,7 +126,6 @@ def __init__(self, leftMotor, rightMotor):
 
     def arcadeDrive(self, xSpeed, zRotation, squaredInputs=True):
         """Arcade drive method for differential drive platform.
-        The calculated values will be squared to decrease sensitivity at low speeds
```
Comment (twinters007) [inline wpilib/wpilib/drive/differentialdrive.py]: What are you doing?

File: wpilib/wpilib/drive/robotdrivebase.py [hunk 4]
```diff
@@ -52,24 +52,23 @@ def __init__(self):
...
 
-    def setDeadband(self, deadband):
-        """Change the value for deadband scaling. The default value is `.02`. Values smaller than the
-         deadband are set to 0, while values larger than the deadband are scaled from 0.0 to 1.0. See
-         :meth:`applyDeadband()`
-
-         :param deadband: The deadband to set
-         :type deadband: float
-         """
+    def setDeadband(self, deadband: float):
```
Comment (twinters007) [inline wpilib/wpilib/drive/robotdrivebase.py]: Why here and not everywhere else

File: wpilib/wpilib/drive/robotdrivebase.py [hunk 5]
```diff
@@ -52,24 +52,23 @@ def __init__(self):
...
+        """
         self.deadband = deadband
 
-    def setMaxOutput(self, maxOutput):
+    def setMaxOutput(self, maxOutput: float):
         """Configure the scaling factor for using drive methods with motor controllers in a mode
         other than PercentVbus or to limit the maximum output.
 
-        The default value is `1.0`
+        The default value is :data:`kDefaultMaxOutput`.
```
Comment (twinters007) [inline wpilib/wpilib/drive/robotdrivebase.py]: In my (limited) experience, this doesn't resolve and just shows "kDefaultMaxOutput". Am I wrong?

File: wpilib/wpilib/drive/killoughdrive.py [hunk 6]
```diff
@@ -1,4 +1,11 @@
 # validated: 2017-12-25 TW f9bece2ffbf7 edu/wpi/first/wpilibj/drive/KilloughDrive.java
+#----------------------------------------------------------------------------
+# Copyright (c) FIRST 2017. All Rights Reserved.
```
Comment (twinters007) [inline wpilib/wpilib/drive/killoughdrive.py]: They've updated all of the copyrights to be x-2018. So this should be 2017-2018

File: wpilib/wpilib/drive/mecanumdrive.py [hunk 7]
```diff
@@ -1,4 +1,11 @@
 # validated: 2017-12-25 TW f9bece2ffbf7 edu/wpi/first/wpilibj/drive/MecanumDrive.java
+#----------------------------------------------------------------------------
+# Copyright (c) FIRST 2008-2017. All Rights Reserved.
```
Comment (twinters007) [inline wpilib/wpilib/drive/mecanumdrive.py]: Same thing

File: wpilib/wpilib/drive/differentialdrive.py (hunk 1 above)
Comment (auscompgeek) [reply wpilib/wpilib/drive/differentialdrive.py]: This is what upstream has.

File: wpilib/wpilib/drive/differentialdrive.py (hunk 3 above)
Comment (auscompgeek) [reply wpilib/wpilib/drive/differentialdrive.py]: This isn't accurate if you pass in `squaredInputs=False`.

File: wpilib/wpilib/drive/robotdrivebase.py (hunk 5 above)
Comment (auscompgeek) [reply wpilib/wpilib/drive/robotdrivebase.py]: You are indeed wrong. It links fine.

File: wpilib/wpilib/drive/robotdrivebase.py (hunk 4 above)
Comment (auscompgeek) [reply wpilib/wpilib/drive/robotdrivebase.py]: I don't see anywhere else in this file that has type information. Too lazy to add in the missing types.

File: wpilib/wpilib/drive/differentialdrive.py (hunk 2 above)
Comment (auscompgeek) [reply wpilib/wpilib/drive/differentialdrive.py]: It also made the docs not that pretty. Given there's a note right above it, I don't think it's very useful nor practical to list it again here.

File: wpilib/wpilib/drive/differentialdrive.py (hunk 3 above)
Comment (twinters007) [reply wpilib/wpilib/drive/differentialdrive.py]: Make a PR upstream?

File: wpilib/wpilib/drive/robotdrivebase.py (hunk 5 above)
Comment (twinters007) [reply wpilib/wpilib/drive/robotdrivebase.py]: Does it just link or will it replace it with '1.0'

Comment (auscompgeek) [reply wpilib/wpilib/drive/robotdrivebase.py]: It doesn't show the value. Although I just realised the correct directive would be `:const:`, but not even that shows the value by default. We can easily show the value instead, but we'd have to remember to change it if it changes.

File: wpilib/wpilib/drive/differentialdrive.py (hunk 1 above)
Comment (twinters007) [reply wpilib/wpilib/drive/differentialdrive.py]: Why can't I read?

File: wpilib/wpilib/drive/differentialdrive.py (hunk 3 above)
Comment (auscompgeek) [reply wpilib/wpilib/drive/differentialdrive.py]: Upstream doesn't have this for `arcadeDrive(xSpeed, zRotation, squaredInputs)`.

Comment (twinters007) [reply wpilib/wpilib/drive/differentialdrive.py]: Oh it's just for the default...I see

**Pull request merged**